packages/edit-site/src/components/page-templates/template-actions.js

```diff
@@ -89,29 +112,82 @@ export const deleteTemplateAction = {
 	isPrimary: true,
 	icon: trash,
 	isEligible: isTemplateRemovable,
+	supportsBulk: true,
 	hideModalHeader: true,
-	RenderModal: ( { item: template, closeModal } ) => {
+	RenderModal: ( { items: templates, closeModal, onPerform } ) => {
 		const { removeTemplate } = useDispatch( editSiteStore );
+		const { createSuccessNotice, createErrorNotice } =
+			useDispatch( noticesStore );
+		const { deleteEntityRecord } = useDispatch( coreStore );
 		return (
 			<VStack spacing="5">
 				<Text>
-					{ sprintf(
-						// translators: %s: The template or template part's title.
-						__( 'Are you sure you want to delete "%s"?' ),
-						decodeEntities( template.title.rendered )
-					) }
+					{ templates.length > 1
+						? sprintf(
+								// translators: %s: The template or template part's title.
+								__(
+									'Are you sure you want to delete %s items?'
+								),
+								decodeEntities( templates.length )
```

Comment:
No need for `decodeEntities` here. This also applies [above](https://github.com/WordPress/gutenberg/pull/57255/files#diff-a147bc579639fdc67bfc7fbc2b5a319cbb46f90a99430ac5c0005e58ce390dbdR64).